content/how-to/author-libraries.md

```diff
@@ -1,7 +1,172 @@
 ---
 title: How to Author Libraries?
+contributors:
+    - pksjce
 ---
-> library option
-> externals option
 
-> see also [[Output]]
+webpack is a tool which can be used to bundle application code and also to bundle library code. If you are the author of JavaScript library and are looking to streamline your bundle strategy then this document will help you.
+
+## Author a library
+
+We have here a small wrapper library to convert number 1 to 5 from number to word and vice-versa. It looks something like this.
+
+__index.js__
+```javascript
+import _ from 'lodash';
+import numRef from './ref.json';
+
+function createTransalator() {
+    return {
+        numtoword: (num) => {
+            return num < 0 || num > 5 ? 'This is a failure' : converttoword(num);
+        },
+        wordtonum: (word) => {
+            const num = converttonum(word);
+            return num === -1 ? 'This is a failure' : num;
+        }
+    };
+}
+
+const converttoword = (num) => {
+    return _.reduce(numRef, (accum, ref) => {
+        return ref.num === num ? ref.word : accum;
+    }, '');
+};
+
+const converttonum = (word) => {
+    return _.reduce(numRef, (accum, ref) => {
+        return ref.word === word && word.toLowerCase() ? ref.num : accum;
+    }, -1);
+};
+
+module.exports = createTransalator();
+
+```
+For full library configuration and code please refer to [webpack-library-example](https://github.com/kalcifer/webpack-library-example)
+
+## Configure webpack
+
+Now the agenda is to bundle this library
+  - Without bundling lodash but requiring it to be loaded by the consumer.
+  - Name of the library is `webpack-numbers` and the variable is `webpackNumbers`.
+  - Library can be imported as `import webpackNumbers from 'webpack-numbers'` or `require('webpack-numbers')`.
+  - Library can be accessed through global variable `webpackNumbers` when included through `script` tag.
+  - Library can be accessed inside nodejs.
+
+### Add webpack
+
+Add basic webpack configuration.
+Also add a [`.babelrc`](https://babeljs.io/docs/usage/babelrc/) file so that you can write webpack config in es2015 syntax.
+
+__webpack.config.babel.js__
+```javascript
+export default () => (
```

Comment:
Using objects is a bit more consistent with the other pages, but it doesn't really matter. A few of the examples are invalid though, `export default = () => ...` doesn't work.